src/test/java/com/lgtm/easymoney/controllers/RequestControllerTest.java

```diff
@@ -0,0 +1,366 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Unit test for TransferController.
+ * {@link com.lgtm.easymoney.controllers.RequestController}
+ * */
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(RequestController.class)
+@ContextConfiguration
+public class RequestControllerTest {
+  @Autowired
+  private MockMvc mvc;
+  @MockBean
+  private RequestService requestService;
+
+  private RequestReq requestReq;
+
+  private RequestRsp requestRsp;
+  private ResourceCreatedRsp resourceCreatedRsp;
+  private TransactionRsp transactionRsp;
+
+  private Long fromUid = 1L;
+  private Long toUid = 2L;
+
+  private Long requestId = 1L;
+  private BigDecimal amount = BigDecimal.valueOf(30.0);
+  private String description = "this is a test request";
+  private Date lastUpdateTime = new Date(20221020L);
+
+  /**
+   * Set up reusable test fixtures.
+   * */
+  @Before
+  public void setUp() {
+    // request for making request
+    requestReq = new RequestReq();
+    requestReq.setFromUid(fromUid);
+    requestReq.setToUid(toUid);
+    requestReq.setAmount(amount);
+    requestReq.setCategory(Category.PARTY);
+    requestReq.setDescription(description);
+    // transaction response
+    transactionRsp = new TransactionRsp(fromUid, toUid, amount,
+            TransactionStatus.TRANS_COMPLETE, description, Category.PARTY, lastUpdateTime);
+    // response for making requests
+    resourceCreatedRsp = new ResourceCreatedRsp(1L);
+    requestRsp = new RequestRsp();
+    requestRsp.setSuccess(true);
+    requestRsp.setCurrBalance(BigDecimal.valueOf(70.0));
+    requestRsp.setRequests(List.of(transactionRsp));
+    // transaction
+  }
+
+  @Test
+  public void createRequestSuccess() throws Exception {
+    // Arrange
+    Mockito.when(requestService.createRequest(requestReq)).thenReturn(resourceCreatedRsp);
+
+    // Act
+    ResultActions returnedResponse = postRequest(requestReq);
+
+    // Assert
+    returnedResponse.andExpectAll(
+            status().isOk(),
+            jsonPath("$.id").value(1L));
+  }
+
+  @Test
+  public void createRequestFailedWithNegativeAmount() throws Exception {
+    // Arrange
+    requestReq.setAmount(BigDecimal.valueOf(-100));
+    Mockito.when(requestService.createRequest(requestReq)).thenReturn(resourceCreatedRsp);
```

Comment:
The validation exception is thrown before invoking service, so I think it's safe to remove the mock here